avocado/core/runners/avocado_instrumented.py

```diff
@@ -39,17 +40,28 @@ def _run_avocado(runnable, queue):
 
             klass, method = klass_method.split('.', 1)
 
-            # inject the run.test_parameters option in the Test params
-            # this handles the -p command-line argument
-            params = TreeNode().get_node("/", True)
-            params.value = dict(runnable.config.get('run.test_parameters', []))
+            # inject the -p command-line / run.test_parameters option
```

Comment:
I think it will be better to move parameters creation into its own method, for better readability.

Reply:
> > @clebergnu @beraldoleal, FYI, while I was investigating #4840, I found a problem in this code:
> > ```
> > [wrampazz@wrampazz avocado.dev]$ avocado-runner-avocado-instrumented runnable-run -u /tmp/foo.py:First.test_pass
> > {'status': 'started', 'time': 64065.782545549}
> > {'type': 'stderr', 'log': b'Traceback (most recent call last):\n  File "/home/wrampazz/src/avocado/avocado.dev/avocado/core/runners/avocado_instrumented.py", line 45, in _run_avocado\n    for path, env in runnable.variant[\'variant\']]\nTypeError: \'NoneType\' object is not subscriptable\n', 'encoding': 'utf-8', 'status': 'running', 'time': 64065.786519159}
> > {'result': 'error', 'fail_reason': "'NoneType' object is not subscriptable", 'status': 'finished', 'time': 64065.786647994}
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I changed the logic inside `avocado/core/runners/avocado_instrumented.py`, the last commit, to avoid the problem.
> 
> Not sure if I understood, but the error on #4840 seems to be different. Using this patch the problem there is still there. I'm debugging there and it is related to the test path resolution.

The error of #4840 still exists. What I meant was when I was debugging #4840, I was on the branch of this PR and it gave me the following error, which is related to the code of this PR:

```
"/home/wrampazz/src/avocado/avocado.dev/avocado/core/runners/avocado_instrumented.py", line 45, in _run_avocado\n    for path, env in runnable.variant[\'variant\']]\nTypeError: \'NoneType\' object is not subscriptable\n'
```

I changed the logic to avoid the problem above.

Reply:
Do you mean to create a new method for the following 3 lines returning the `paths` and the `tree_nodes`? If that is the case, it is also worth adding a method to create the variant tree. I can do that, but I don't know if it will improve that much the readability. Let me know.

Reply:
No, I meant the whole part for creating `tree_nodes` and `path` the parameters part and also creating the variant tree. lines 43-57.

Reply:
Okay, got it. I'll change that.

Reply:
Done @richtja! Thanks for the review.